lib/ansible/executor/task_executor.py

```diff
@@ -476,16 +477,16 @@ def _execute(self, variables=None):
                 self._play_context.remote_addr != self._connection._play_context.remote_addr):
             self._connection = self._get_connection(variables=variables, templar=templar)
             hostvars = variables.get('hostvars', None)
-            if hostvars:
+            # only template the vars if the connection actually implements set_host_overrides
+            # NB: this is expensive, and should be removed once connection-specific vars are being handled by play_context
+            if hostvars and getattr(self._connection, 'set_host_overrides') != getattr(ConnectionBase, 'set_host_overrides'):
```

Comment:
Any reason you are using getattr instead of just accessing the method directly?  getattr has no default, and unless a default is supplied explicitly, a getattr will raise an AttributeError if the method is missing.